### PR TITLE
[Bugfix:TAGrading] peer grading subtotal/total

### LINKED ISSUE #5471: Totals & Subtotals while Grading (from TA & Peer View)
Just as a reminder...  this is what the rubric panel  looks like when a TA is grading a Autograde + Manual Grading (but no peer grading) assignment:  ( this shouldn't change )

autograding+tagrading

When a student/peer is grading (their grading is incomplete), this is what they should see:

partial_peer_grading

When the student/peer is finished grading, this is what they should see:
Note: the student/peer should never see any information about the autograding total, or the TA scores, or the other peer scores, or the overall Total.  
The peer can see the autograding panel with the autograding test cases, but we won't include the autograding score in the rubric panel. 

completed_peer_grading

This is what a TA should see for an assignment with both non-peer and peer Manual grading:
Note: normally TAs do not complete the peer components/marks.
If the assignment includes Autograding, those lines & scores are included as indicated.

no_ta_peer_grade

If the TA does enter their own scores for the peer components/marks, it will look like this:

ta_did_peer_grading

@emaicus @jchulton @josharsh    Comments/Questions?

Also tagging @mukul-kmr-jha  -- who is refactoring this webpage panel layout system

### PR DESCRIPTION
**Pull request by josharsh**
### Please check if the PR fulfills these requirements:
closes #5471 
* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the new behavior?
Corrects stats on TA, peer panels as mentioned in the issue #5471

### CODE REVIEW

**Comment by jchulton** [comment]
This should be 0/10 Combined Peer Grading Total (the screenshot is for an instructor in which peer grading has not been done):
image

**Comment by jchulton** [comment]
image
The TA rubric for an instructor is missing the various stats described in #5471

**Comment by josharsh** [comment]
@jchulton Remember the other day we were talking about Twig Files caching up. I think that's what has been happening when you are seeing those stats on the screenshots you posted. (I don't know but caching actually happens with component level twig files)
I think so because there is this term "Manual Grading Points" which does not exist in the updated work and when I tried to regenerate this was the screen. Please tell if I am missing something. 
image

File: site/app/models/gradeable/GradedComponentContainer.php [hunk 1]
```diff
@@ -226,6 +226,37 @@ public function getTotalScore(User $grader = null) {
...
+     */
+    public function getTotalTaScore(User $grader = null) {
+        $points_earned = 0.0;
+        $number_of_graders = 0;
+        // TODO: how should peer grades be calculated: now its an average
+        /** @var GradedComponent $graded_component */
+        foreach ($this->graded_components as $graded_component) {
+            // If there is a grader, we are only computing their total score rather than the total score for all peers.
+            if ($graded_component->getGrader()->getGroup() < User::GROUP_STUDENT) {
+                if($graded_component->isPeer()){
```
[the remark this answers is not in the document]
Comment (josharsh) [reply site/app/models/gradeable/GradedComponentContainer.php:242]: Corrected!

File: site/public/js/ta-grading-rubric.js [hunk 2]
```diff
@@ -1322,7 +1358,29 @@ function getTaGradingComplete() {
...
 function getTaGradingTotal() {
     let total = 0.0;
-    $('.component').each(function () {
+    $('.ta-component').each(function () {
+        total += parseFloat($(this).attr('data-max_value'));
+    });
+    return total;
+}
+/**
+ * Gets the number of ta grading points that can be earned
```
[the remark this answers is not in the document]
Comment (josharsh) [reply site/public/js/ta-grading-rubric.js]: Yes, Corrected

File: site/public/templates/grading/TotalScoreBox.twig [hunk 3]
```diff
@@ -5,34 +5,65 @@ Required Parameters:
...
+-user_group
+l
 #}
-{% set show_total = auto_grading_total is defined and ta_grading_total is defined %}
-{% if auto_grading_total is defined %}
-    <div class="box-title badge-container">
-        <strong id="autograding_total" class="badge {{ functions.getBadgeStyle(auto_grading_earned, auto_grading_total, auto_grading_complete) }}">
-            {{ auto_grading_earned is defined ? auto_grading_earned|round(decimal_precision) : '&minus;' }} / {{ auto_grading_total|round(decimal_precision) }}
+{% if user_group < 4 %}
+    {% set show_total = auto_grading_total is defined and ta_grading_total is defined %}
```
Comment (jchulton) [inline site/public/templates/grading/TotalScoreBox.twig:13]: Total should be shown even if there is no autograding. I think it should always be shown for any non-peer grader.

Comment (josharsh) [reply site/public/templates/grading/TotalScoreBox.twig]: Updated this!

**Comment by jchulton** [comment]
Getting /NaN for Total 
image

**Comment by josharsh** [comment]
@jchulton Not able to reproduce this in development course ->both peer and non peer gradeables 
image
Could you please tell the steps you took?

**Comment by jchulton** [comment]
I just created a new gradeable with some simple components and marks, submitted as instructor for a student pushed it down to grades released and graded it. Not sure why it's not repeatable...

**Comment by emaicus** [comment]
I was also able to reproduce the NaN bug. 
Steps: 
1. Create a new gradeable. 
2. Create two new 5 point components, one by count up, the other by count down.
3. Create a set of marks for the components (I created the same ones as @jchulton's above screenshot)
4. Make a submission for a user (I used aphacker).
5. Grade the submission.

**Comment by josharsh** [comment]
Hi @jchulton @emaicus The problem was I was not creating a new gradeable and was testing on existing gradeables which had autograding enabled. Now the total score (being shown as NaN) was a sum of autograding and manual grading total and since autograding was not defined, it was showing NaN. I have put in checks for the same.

**Comment by jchulton** [comment]
The total score is now showing properly, but the badge is always green: should be red below
image

**Comment by jchulton** [comment]
As per #5471, the Combined Peer Grading Total should appear on the Grading panel as well as the Peer Grading panel. Also the Total should be calculated using the Combined Peer Grading Total instead of the Individual Peer Grading Total, as below:
image

**Review by shailpatels — CHANGES_REQUESTED** [review]
Don't think this is working 

image

On the instructor side all the grade totals seem incorrect I can't get it past 0/0 unless I'm doing something wrong?

**Comment by emaicus** [comment]
Superseded by #5982

**Pull request closed without merging**